Codigo/GameLogic.bas

```diff
@@ -2135,8 +2135,77 @@ Public Sub TimerQuestOrco()
     Call SendData(SendTarget.ToAll, 0, PrepareMessagePlayWave(156, NO_3D_SOUND, NO_3D_SOUND))
 End Sub
 
+Public Function TestRequiredEquipedItem(ByRef inventory As t_Inventario, ByVal requiredItemsFlag As Long) As e_EquipedSlotMask
+    If IsSet(requiredItemsFlag, e_EquipedSlotMask.eArmor) And _
+      inventory.ArmourEqpObjIndex = 0 Then
+        TestRequiredEquipedItem = e_EquipedSlotMask.eArmor
+        Exit Function
+    End If
+    If IsSet(requiredItemsFlag, e_EquipedSlotMask.eHelm) And _
+      inventory.CascoEqpObjIndex = 0 Then
+        TestRequiredEquipedItem = e_EquipedSlotMask.eHelm
+        Exit Function
+    End If
+    If IsSet(requiredItemsFlag, e_EquipedSlotMask.eKnucle) And _
+      inventory.NudilloObjIndex = 0 Then
+        TestRequiredEquipedItem = e_EquipedSlotMask.eKnucle
+        Exit Function
+    End If
+    If IsSet(requiredItemsFlag, e_EquipedSlotMask.eMagicItem) And _
+      inventory.MagicoObjIndex = 0 Then
+        TestRequiredEquipedItem = e_EquipedSlotMask.eMagicItem
+        Exit Function
+    End If
+    If IsSet(requiredItemsFlag, e_EquipedSlotMask.eProjectile) And _
+      inventory.MunicionEqpObjIndex = 0 Then
+        TestRequiredEquipedItem = e_EquipedSlotMask.eProjectile
+        Exit Function
+    End If
+    If IsSet(requiredItemsFlag, e_EquipedSlotMask.eShield) And _
+      inventory.EscudoEqpObjIndex = 0 Then
+        TestRequiredEquipedItem = e_EquipedSlotMask.eShield
+        Exit Function
+    End If
+    If IsSet(requiredItemsFlag, e_EquipedSlotMask.eShip) And _
+      inventory.BarcoObjIndex = 0 Then
+        TestRequiredEquipedItem = e_EquipedSlotMask.eShip
+        Exit Function
+    End If
+    If IsSet(requiredItemsFlag, e_EquipedSlotMask.eTool) And _
+      inventory.HerramientaEqpObjIndex = 0 Then
+        TestRequiredEquipedItem = e_EquipedSlotMask.eTool
+        Exit Function
+    End If
+    If IsSet(requiredItemsFlag, e_EquipedSlotMask.eWeapon) And _
+      inventory.WeaponEqpObjIndex = 0 Then
+        TestRequiredEquipedItem = e_EquipedSlotMask.eWeapon
+        Exit Function
+    End If
+    TestRequiredEquipedItem = e_EquipedSlotMask.eNone
+End Function
 
-
+Public Sub SendrequiredItemMessage(ByVal UserIndex As Integer, ByVal itemMask As e_EquipedSlotMask, ByVal message As String)
+    Select Case itemMask
+        Case e_EquipedSlotMask.eArmor
+             Call WriteConsoleMsg(UserIndex, "Necesitás una armadura " + message, e_FontTypeNames.FONTTYPE_INFO)
```

Comment:
Any chance of moving the string `"Necesitás una armadura` to the client so that we can facilitate working on the game translation to other languages ?

So here you would send just a message ID for `requiredItemMessage` and the client would have multiple localized files for that string

Reply:
not with the current translation system we cannot compose this kind of strings, we can pass the second part as parameter but then the we will have one half translated and the second half with the original value